apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TestLogger.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using Google.Cloud.Spanner.V1.Logging;
+using Xunit.Abstractions;
+
+namespace Google.Cloud.Spanner.Data.IntegrationTests
+{
+    internal class TestLogger : DefaultLogger
+    {
+        private static Lazy<TestLogger> s_Instance = new Lazy<TestLogger>(
```

Comment:
I'm definitely being pedantic now, so apologies, but I believe you actually can predict when they will run (see https://msdn.microsoft.com/en-us/library/aa645758(VS.71).aspx). It is static initializers in the absence of a static constructor which cannot be predicted, but since that is the case we're in here anyway, that's probably what you meant.

However, I still don't see the point. The constructor has no side-effects and as I said, the instance uses almost no memory, so at some implementation-defined point, you either (1) allocate the `Lazy<T>` with three reference fields in it which allows delaying the creation of a `TestLogger` or (2) you allocate a `TestLogger` with no fields. Once again, apologies for being pedantic. Feel free to ignore me.